backends/imgui_impl_osx.mm

```diff
@@ -710,3 +755,321 @@ bool ImGui_ImplOSX_HandleEvent(NSEvent* event, NSView* view)
 
     return false;
 }
+
+static void ImGui_ImplOSX_AddTrackingArea(NSView* _Nonnull view)
+{
+    // If we want to receive key events, we either need to be in the responder chain of the key view,
+    // or else we can install a local monitor. The consequence of this heavy-handed approach is that
+    // we receive events for all controls, not just Dear ImGui widgets. If we had native controls in our
+    // window, we'd want to be much more careful than just ingesting the complete event stream.
+    // To match the behavior of other backends, we pass every event down to the OS.
+    if (g_Monitor)
+        return;
+    NSEventMask eventMask = 0;
+    eventMask |= NSEventMaskMouseMoved | NSEventMaskScrollWheel;
+    eventMask |= NSEventMaskLeftMouseDown | NSEventMaskLeftMouseUp | NSEventMaskLeftMouseDragged;
+    eventMask |= NSEventMaskRightMouseDown | NSEventMaskRightMouseUp | NSEventMaskRightMouseDragged;
+    eventMask |= NSEventMaskOtherMouseDown | NSEventMaskOtherMouseUp | NSEventMaskOtherMouseDragged;
+    eventMask |= NSEventMaskKeyDown | NSEventMaskKeyUp | NSEventMaskFlagsChanged;
+    g_Monitor = [NSEvent addLocalMonitorForEventsMatchingMask:eventMask
```

Comment:
Could you please clarify reason for expanding monitored events list? I think this handles inputs for viewport windows, just wanted to be sure.

Reply:
I didn't write the code to expand the monitored event list originally – it was already there and is probably not necessary.

Reply:
It was there, but event list was way shorter (only `NSEventMaskKeyUp | NSEventMaskFlagsChanged`). It seems you have moved event handling from example app into backend itself. Good cleanup, apple examples are a bit too cluttered with platform stuff :+1: